### PR TITLE
Config is now initialized and passed through main.ts

### PR DESCRIPTION
**Pull request by DebitCardz**

### CODE REVIEW

**Comment by DebitCardz** [comment]
Also added `DiscordConfigSection` and `GithubConfigSection` so when the time comes we can just pass those for wherever we need.

File: src/types/Config.ts [hunk 1]
```diff
@@ -1,21 +1,21 @@
 export default interface Config {
-  discord: {
-    secret: string;
-    channels: {
-      logs: string;
-      issues: {
-        parent?: string;
-        name: string;
-      };
+  discord: DiscordConfigSection;
```
Comment (CoreyShupe) [inline src/types/Config.ts:2]: Can we rename this and the github one to `DiscordConfiguration` and `GithubConfiguration`

**Review by CoreyShupe — CHANGES_REQUESTED** [review]

File: src/types/Config.ts (hunk 1 above)
Comment (DebitCardz) [reply src/types/Config.ts:2]: Yeah sure, the inner Spigot developer in me just told me to name it Section.

**Comment by DebitCardz** [comment]
Hold on integration/github.ts is throwing an error because of `Uncaught ReferenceError: Cannot access 'config' before initialization`

**Comment by CoreyShupe** [comment]
This should be handled better via #12

**Pull request closed without merging**